Firestore/Swift/Source/AsyncAwait/Firestore+AsyncAwait.swift

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+import Foundation
+
+#if compiler(>=5.5) && canImport(_Concurrency)
+  @available(iOS 15, tvOS 15, macOS 12, watchOS 8, *)
+  public extension Firestore {
+    /// Loads a Firestore bundle into the local cache.
+    /// - Parameter bundleData: Data from the bundle to be loaded.
+    /// - Returns: The final `LoadBundleTaskProgress` that contains the total number of documents loaded.
```

Comment:
Nit: can you also describe what type of error is thrown, and the conditions that may cause it to fail? https://developer.apple.com/documentation/xcode/writing-symbol-documentation-in-your-source-files#Describe-the-Thrown-Errors-of-a-Method

Reply:
Done - but I am not quite sure how to specify what type of error we throw since we just throw an `NSError` with a custom error domain.